src/stan/mcmc/hmc/xhmc/adapt_dense_e_xhmc.hpp

```diff
@@ -0,0 +1,55 @@
+#ifndef STAN_MCMC_HMC_xhmc_ADAPT_DENSE_E_XHMC_HPP
+#define STAN_MCMC_HMC_xhmc_ADAPT_DENSE_E_XHMC_HPP
+
+#include <stan/interface_callbacks/writer/base_writer.hpp>
+#include <stan/mcmc/stepsize_covar_adapter.hpp>
+#include <stan/mcmc/hmc/xhmc/dense_e_xhmc.hpp>
+
+namespace stan {
+  namespace mcmc {
+    /**
+      * The No-U-Turn sampler (XHMC) with multinomial sampling
```

Comment:
Is there a top-level note about the different flavors of NUTS in play?  I'm not sure what XHMC is (I was guessing it was exhaustive from your papers, but here it's just listed as being a name for NUTS.